solidity/test/integration/AcreMultiAssetVault.test.ts

```diff
@@ -0,0 +1,200 @@
+import { helpers, ethers } from "hardhat"
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers"
+import { expect } from "chai"
+import {
+  loadFixture,
+  setBalance,
+} from "@nomicfoundation/hardhat-toolbox/network-helpers"
+
+import { ContractTransactionResponse } from "ethers"
+
+import { IMezoPortal, AcreMultiAssetVault, IERC20 } from "../../typechain"
+
+import { to1e18 } from "../utils"
+import { expectedMainnetAddresses, integrationTestFixture } from "./helpers"
+
+const { impersonateAccount } = helpers.account
+
+const testData = {
+  SolvBtc: {
+    assetAddress: expectedMainnetAddresses.solvBtc,
+    assetHolderAddress: "0x19b5cc75846BF6286d599ec116536a333C4C2c14",
+    depositAmount: to1e18(100),
+  },
+  SolvBtcBbn: {
+    assetAddress: expectedMainnetAddresses.solvBtcBbn,
+    assetHolderAddress: "0x8bc93498b861fd98277c3b51d240e7E56E48F23c",
+    depositAmount: to1e18(1000),
+  },
+}
+
+describe("AcreMultiAssetVault", () => {
+  let multiAssetVault: AcreMultiAssetVault
+  let mezoPortal: IMezoPortal
+
+  before(async () => {
+    ;({ multiAssetVault, mezoPortal } = await loadFixture(
+      integrationTestFixture,
+    ))
+  })
+
+  const testRoundTrip = ({
+    assetAddress,
+    assetHolderAddress,
+    depositAmount,
+  }: {
+    assetAddress: string
+    assetHolderAddress: string
+    depositAmount: bigint
+  }) =>
+    // eslint-disable-next-line func-names
+    function () {
+      describe("round trip", () => {
+        let asset: IERC20
+
+        let expectedAcreDepositId: bigint
+        let expectedMezoDepositId: bigint
+
+        let depositor: HardhatEthersSigner
+        let depositOwner: HardhatEthersSigner
+        let receiver: HardhatEthersSigner
+
+        before(async () => {
+          asset = await ethers.getContractAt("IERC20", assetAddress)
+
+          await impersonateAccount(assetHolderAddress)
+          await setBalance(assetHolderAddress, to1e18(1))
```

Comment:
Not for this PR but I would consider renaming `setBalance` to `setEthBalance`. For a moment I was confused if this is Solv BTC balance and if so, why do we allow it to be lower than the deposit amount from the test data.

Reply:
Agree, but this is a function imported from `@nomicfoundation/hardhat-toolbox/network-helpers`.
We can try to wrap it up with our hardhat helpers.